integration-cli/docker_cli_build_test.go

```diff
@@ -4694,3 +4694,90 @@ func TestBuildDockerfileOutsideContext(t *testing.T) {
 
 	logDone("build - Dockerfile outside context")
 }
+
+func TestBuildSpaces(t *testing.T) {
+	// Test to make sure that leading/trailing spaces on a command
+	// doesn't change the error msg we get
+	var (
+		err1 error
+		err2 error
+	)
+
+	name := "testspaces"
+	defer deleteImages(name)
+	ctx, err := fakeContext("FROM busybox\nRUN\n",
+		map[string]string{
+			"Dockerfile": "FROM busybox\nRUN\n",
+		})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ctx.Close()
+
+	if _, err1 = buildImageFromContext(name, ctx, false); err1 == nil {
+		t.Fatal("Build 1 was supposed to fail, but didn't")
+	}
+
+	ctx.Add("Dockerfile", "FROM busybox\nRUN    ")
+	if _, err2 = buildImageFromContext(name, ctx, false); err2 == nil {
+		t.Fatal("Build 2 was supposed to fail, but didn't")
+	}
+
+	// Skip over the times
+	e1 := err1.Error()[strings.Index(err1.Error(), `level="`):]
+	e2 := err2.Error()[strings.Index(err1.Error(), `level="`):]
+
+	if strings.Replace(e1, " ", "", -1) != strings.Replace(e2, " ", "", -1) {
```

Comment:
If I'm remembering correctly, the error message printed the original line from the Dockerfile in the error message. So, the two error messages were the same except for the differences in the spaces - so I'm, in essence, masking the spaces.